src/rgw/driver/rados/config/zone.cc

```diff
@@ -163,31 +163,90 @@ class RadosZoneWriter : public sal::ZoneWriter {
     return 0;
   }
 }; // RadosZoneWriter
+ 
 
+int RadosConfigStore::get_zones_pool_set(const DoutPrefixProvider *dpp,
+                       optional_yield y,
+                       std::string_view my_zone_id,
+                       std::set<rgw_pool>& pools)
```

Comment:
while it's reasonable to think that this 'pool' stuff should be rados-specific, it's really just making sure the pool name strings are unique across all of the `ConfigStore`'s zone definitions

non-rados backends may choose to rely on the pool name strings to avoid naming conflicts between different storage types (like lifecycle vs. gc) or between zones

and it's not `RadosConfigStore` that depends on these pool names, but `RadosStore`; and nothing prevents us from configuring a `DBConfigStore` with a `RadosStore` or vice versa

so i think this needs to stay 'above the line' so that all `ConfigStore` backends initialize their zones with unique pool names, whether or not their backend store will use them